openedx/core/djangoapps/user_api/preferences/tests/test_api.py

```diff
@@ -0,0 +1,324 @@
+# -*- coding: utf-8 -*-
+"""
+Unit tests for preference APIs.
+"""
+import datetime
+import ddt
+import unittest
+from mock import patch
+from pytz import UTC
+
+from django.conf import settings
+from django.contrib.auth.models import User
+from django.test import TestCase
+from django.test.utils import override_settings
+from dateutil.parser import parse as parse_datetime
+
+from student.tests.factories import UserFactory
+
+from xmodule.modulestore.tests.django_utils import ModuleStoreTestCase
+from xmodule.modulestore.tests.factories import CourseFactory
+
+from ...accounts.api import create_account
+from ...errors import UserNotFound, UserNotAuthorized, PreferenceValidationError, PreferenceUpdateError
+from ...models import UserPreference, UserProfile, UserOrgTag
+from ...preferences.api import (
+    get_user_preference, get_user_preferences, set_user_preference, update_user_preferences, delete_user_preference,
+    update_email_opt_in
+)
+
+
+@unittest.skipUnless(settings.ROOT_URLCONF == 'lms.urls', 'Account APIs are only supported in LMS')
+class TestPreferenceAPI(TestCase):
+    """
+    These tests specifically cover the parts of the API methods that are not covered by test_views.py.
+    This includes the specific types of error raised, and default behavior when optional arguments
+    are not specified.
+    """
+    password = "test"
+
+    def setUp(self):
+        super(TestPreferenceAPI, self).setUp()
+        self.user = UserFactory.create(password=self.password)
+        self.different_user = UserFactory.create(password=self.password)
+        self.staff_user = UserFactory(is_staff=True, password=self.password)
+        self.no_such_user = UserFactory.create(password=self.password)
+        self.no_such_user.username = "no_such_user"
+        self.test_preference_key = "test_key"
+        self.test_preference_value = "test_value"
+        UserPreference.set_preference(self.user, self.test_preference_key, self.test_preference_value)
+
+    def test_get_user_preference(self):
+        """
+        Verifies the basic behavior of get_user_preference.
+        """
+        self.assertEqual(
```

Comment:
Funny, I was thinking about that as I was writing these. I did some quick Googling at the time and it seemed that Python generally goes the opposite way. 

http://stackoverflow.com/questions/2404978/why-are-assertequals-parameters-in-the-order-expected-actual
https://groups.google.com/forum/#!msg/pythoscope/S93XxsTVBY4/8J_ynC8M7xEJ

I've posted a question in the dev room to gauge how others are doing this.